src/swap/types.ts

```diff
@@ -348,3 +356,10 @@ export type Transaction = {
   to: Address; // The recipient address
   value: bigint; // The value of the transaction
 };
+
+export type UseAwaitCallsParams = {
+  accountConfig: Config;
+  config: SwapConfig;
+  lifecycleStatus: LifecycleStatus;
+  setLifecycleStatus: React.Dispatch<React.SetStateAction<LifecycleStatus>>;
```

Comment:
I believe we don't use `setLifecycleStatus` like this anymore. At this point passing around `setLifecycleStatus` is an antipattern we should avoid at all coast.

Right @alessey ?

Reply:
modified to use `updateLifecycleStatus`